src/test/java/com/cdancy/jenkins/rest/features/JobsApiLiveTest.java

```diff
@@ -114,126 +119,257 @@ public void testGetBuildInfo() {
+        assertTrue(output.fullDisplayName().contains("JobInFolder #1"));
+        assertTrue(output.queueId() == queueIdForAnotherJob);
+    }
+
+    @Test(dependsOnMethods = "testGetBuildInfoOfJobInFolder")
+    public void testDeleteJobInFolder() {
+        RequestStatus success = api().delete("job/test-folder", "JobInFolder");
+        assertTrue(success.value());
+    }
+
+    @Test(dependsOnMethods = "testDeleteJobInFolder")
+    public void testDeleteFolder() {
+        RequestStatus success = api().delete(null, "test-folder");
+        assertTrue(success.value());
+    }
+
     @Test
     public void testGetJobInfoNonExistentJob() {
-        JobInfo output = api().jobInfo(randomString());
+        JobInfo output = api().jobInfo(null, randomString());
         assertNull(output);
     }
 
     @Test
     public void testDeleteJobNonExistent() {
-        RequestStatus success = api().delete(randomString());
+        RequestStatus success = api().delete(null, randomString());
         assertNotNull(success);
         assertFalse(success.value());
     }
 
     @Test
     public void testGetConfigNonExistentJob() {
-        String output = api().config(randomString());
+        String output = api().config(null, randomString());
         assertNull(output);
     }
 
     @Test
     public void testSetDescriptionNonExistentJob() {
-        boolean success = api().description(randomString(), "RandomDescription");
+        boolean success = api().description(null, randomString(), "RandomDescription");
         assertFalse(success);
     }
 
     @Test
     public void testGetDescriptionNonExistentJob() {
-        String output = api().description(randomString());
+        String output = api().description(null, randomString());
         assertNull(output);
     }
 
     @Test
     public void testBuildNonExistentJob() {
-        Integer output = api().build(randomString());
+        Integer output = api().build(null, randomString());
         assertNull(output);
     }
 
     @Test
     public void testGetBuildInfoNonExistentJob() {
-        BuildInfo output = api().buildInfo(randomString(), 123);
+        BuildInfo output = api().buildInfo(null, randomString(), 123);
         assertNull(output);
     }
 
     @Test
     public void testBuildNonExistentJobWithParams() {
         Map<String, List<String>> params = new HashMap<>();
         params.put("SomeKey", Lists.newArrayList("SomeVeryNewValue"));
-        Integer output = api().buildWithParameters(randomString(), params);
+        Integer output = api().buildWithParameters(null, randomString(), params);
         assertNull(output);
     }
 
+    private boolean isFolderPluginInstalled() {
+        boolean installed = false;
+        Plugins plugins = api.pluginManagerApi().plugins(3, null);
+        for(Plugin plugin:plugins.plugins()) {
+            if(plugin.shortName().equals(FOLDER_PLUGIN_NAME) && plugin.version().equals(FOLDER_PLUGIN_VERSION)) {
+                installed = true;
```

Comment:
lets put a `break;` after this line so that we don't continue looking through the list of installed plugins which could potentially be very big.